src/app/services/impl/HttpStudentService.ts

```diff
@@ -68,7 +73,24 @@ export class HttpStudentService extends StudentService {
     } else {
       return throwError(new UnauthenticatedError());
     }
+  }
 
+  resetPassword(oldPassword: string, newPassword: string): Observable<boolean> {
+    if (!oldPassword || !newPassword) return of(false);
+    console.log(oldPassword, newPassword);
+    if (!this.hasLogin()) {
+      return of(false);
+    }
+    return this.http.patch<boolean>(`${this.baseUrl}/api/students/${this.currentStudent.id}/password`, {
+      currentPassword: oldPassword,
+      newPassword,
+    }, this.getAuthHeaders()).pipe(catchError(err => {
+      if (err.status === 400) {
+        return throwError(new IncorrectPasswordFoundError());
+      } else {
+        throw throwError(`Catch unknown error from the server: ${err.status}, ${err.toString()}`);
+      }
+    }));
```

Comment:
Don't ever use `true` or `false` to indicate the operation status
if the procedure failed, use `throwError` with a distinguishable and custom `Error` to indicate that.

 **But, it's unnecessary to be too defensive here, since we should have performed Form-Validation**.
In the reset-password's form, the `inputs` must be validated prior to actually invoking the `resetPassword` method.